systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/externalClients/OauthExternalKafkaClient.java

```diff
@@ -0,0 +1,273 @@
+        }
+        vertx.close();
+        return resultPromise;
+    }
+
+    public Future<Integer> receiveMessagesPlain() {
+        return receiveMessagesPlain(Constants.GLOBAL_CLIENTS_TIMEOUT);
+    }
+
+    @Override
+    public Future<Integer> receiveMessagesPlain(long timeoutMs) {
+        String clientName = "receiver-plain-" + clusterName;
+        vertx = Vertx.vertx();
+        CompletableFuture<Integer> resultPromise = new CompletableFuture<>();
+
+        IntPredicate msgCntPredicate = x -> x == messageCount;
+
+        KafkaClientProperties kafkaClientProperties = new KafkaClientProperties.KafkaClientPropertiesBuilder()
+            .withNamespaceName(namespaceName)
+            .withClusterName(clusterName)
+            .withGroupIdConfig(consumerGroup)
+            .withSecurityProtocol("SASL_" + CommonClientConfigs.DEFAULT_SECURITY_PROTOCOL)
+            .withBootstrapServerConfig(getExternalBootstrapConnect(namespaceName, clusterName))
+            .withKeyDeSerializerConfig(StringDeserializer.class.getName())
+            .withValueDeSerializerConfig(StringDeserializer.class.getName())
+            .withClientIdConfig(kafkaUsername + "-consumer")
+            .withAutoOffsetResetConfig("earliest")
+            .withSaslMechanism("OAUTHBEARER")
+            .withSaslLoginCallbackHandlerClass()
+            .withSharedProperties()
+            .withSaslJassConfig(this.clientId, this.clientSecretName, this.oauthTokenEndpointUri)
+            .build();
+
+        vertx.deployVerticle(new Consumer(kafkaClientProperties, resultPromise, msgCntPredicate, topicName, clientName));
+
+        try {
+            resultPromise.get(timeoutMs, TimeUnit.MILLISECONDS);
+        } catch (Exception e) {
+            resultPromise.completeExceptionally(e);
+        }
+        vertx.close();
+        return resultPromise;
+    }
+
+    public Future<Integer> receiveMessagesTls() {
+        return sendMessagesTls(Constants.GLOBAL_CLIENTS_TIMEOUT);
+    }
+
+    @Override
+    public Future<Integer> receiveMessagesTls(long timeoutMs) {
+
+        String clientName = "receiver-ssl-" + clusterName;
+        vertx = Vertx.vertx();
+        CompletableFuture<Integer> resultPromise = new CompletableFuture<>();
+
+        IntPredicate msgCntPredicate = x -> x == messageCount;
+
+        String caCertName = this.caCertName == null ?
+                KafkaResource.getKafkaExternalListenerCaCertName(namespaceName, clusterName) : this.caCertName;
+        LOGGER.info("Going to use the following CA certificate: {}", caCertName);
+
+        KafkaClientProperties kafkaClientProperties = new KafkaClientProperties.KafkaClientPropertiesBuilder()
+            .withNamespaceName(namespaceName)
+            .withClusterName(clusterName)
+            .withCaSecretName(caCertName)
+            .withBootstrapServerConfig(getExternalBootstrapConnect(namespaceName, clusterName))
+            .withKeyDeSerializerConfig(StringDeserializer.class.getName())
```

Comment:
`withKeyDeSerializerConfig` → `withKeyDeserializerConfig`

Also if everything is just going to use `StringDeserializer` why not make that the default so you only need to call the method if you're using a different Deserializer?

Reply:
```
Also if everything is just going to use StringDeserializer why not make that the default so you only need to call the method if you're using a different Deserializer?
```
I can create the next PR. on that.